src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -0,0 +1,308 @@
+            try
+            {
+                response = await ExecuteHookScript<PrepareJobResponse>(context, input, ActionRunStage.Pre, prependPath);
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+                throw new Exception($"Custom container implementation failed with error: {ex.Message}. Please contact your self hosted runner administrator.");
+            }
+            if (response == null)
+            {
+                return;
+            }
+            SaveHookState(context, response.State, input);
+        }
+
+        public async Task RunScriptStepAsync(IExecutionContext context, ContainerInfo container, string entryPointArgs, string entryPoint, IDictionary<string, string> environmentVariables, string prependPath, string workingDirectory)
+        {
+            Trace.Entering();
+            var input = new HookInput
+            {
+                Command = HookCommand.RunScriptStep,
+                ResponseFile = GenerateResponsePath(),
+                Args = new ScriptStepArgs
+                {
+                    EntryPointArgs = entryPointArgs.Split(' ').Select(arg => arg.Trim()),
+                    EntryPoint = entryPoint,
+                    EnvironmentVariables = environmentVariables,
+                    PrependPath = prependPath,
+                    WorkingDirectory = workingDirectory,
+                },
+                State = context.Global.ContainerHookState
+            };
+
+            PrepareJobResponse response;
+            try
+            {
+                response = await ExecuteHookScript<PrepareJobResponse>(context, input, ActionRunStage.Pre, prependPath);
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+                throw new Exception($"Custom container implementation failed with error: {ex.Message}. Please contact your self hosted runner administrator.");
+            }
+            if (response == null)
+            {
+                return;
+            }
+            SaveHookState(context, response.State, input);
+        }
+
+        public string GetContainerHookData()
+        {
+            return JsonUtility.ToString(new { HookIndexPath });
+        }
+
+        private async Task<T> ExecuteHookScript<T>(IExecutionContext context, HookInput input, ActionRunStage stage, string prependPath) where T : HookResponse
+        {
+            var scriptDirectory = Path.GetDirectoryName(HookIndexPath);
+            var stepHost = HostContext.CreateService<IDefaultStepHost>();
+            Dictionary<string, string> inputs = new()
+            {
+                ["standardInInput"] = JsonUtility.ToString(input),
+                ["path"] = HookIndexPath,
+                ["shell"] = HostContext.GetDefaultShellForScript(HookIndexPath, Trace, prependPath)
+            };
+
+            var handlerFactory = HostContext.GetService<IHandlerFactory>();
```

Comment:
feels weird that we are using Handler to execute this, should we use processInvoker directly?